packages/akoya-adapter/src/apiClient.ts

```diff
@@ -0,0 +1,153 @@
+import { enc } from "crypto-js";
+import axios from "axios";
+import type { LogClient, ApiCredentials } from "./models";
+
+export const AKOYA_BASE_PATH = 'https://sandbox-idp.ddp.akoya.com'
+export const AKOYA_BASE_PROD_PATH = 'https://idp.ddp.akoya.com'
+export const AKOYA_PRODUCT_PATH = "https://sandbox-products.ddp.akoya.com"
+export const AKOYA_PRODUCT_PROD_PATH = "https://products.ddp.akoya.com"
+
+function makeAkoyaAuthHeaders(apiConfig: any) {
+  const words = enc.Utf8.parse(`${apiConfig.clientId}:${apiConfig.secret}`);
+  return {
+    Authorization: `Basic ${enc.Base64.stringify(words)}`,
+    "content-type": "application/x-www-form-urlencoded",
+    accept: "application/json",
+  };
+}
+
+function makeAkoyaBearerHeaders(token: string) {
+  return {
+    Authorization: `Bearer ${token}`,
+    accept: "application/json",
+  };
+}
+
+export default class AkoyaClient {
+  apiConfig: ApiCredentials;
+  client_redirect_url: string;
+  authParams: object;
+  envConfig: Record<string, string>;
+  logger: LogClient;
+
+  constructor(sandbox: boolean, apiConfig: ApiCredentials, logger: LogClient, envConfig: Record<string, string>){
+    this.apiConfig = sandbox ? {
+      ...apiConfig,
+      basePath: AKOYA_BASE_PATH,
+      productPath: AKOYA_PRODUCT_PATH,
+      apiVersion: 'v2',
+      aggregator: "akoya_sandbox",
+      available: true,
+    } : {
+      ...apiConfig,
+      basePath: AKOYA_BASE_PROD_PATH,
+      productPath: AKOYA_PRODUCT_PROD_PATH,
+      apiVersion: 'v2',
+      aggregator: "akoya",
+      available: true,
+    };
+    this.logger = logger;
+    this.envConfig = envConfig;
+    this.client_redirect_url = `${this.envConfig.HostUrl}/oauth/${this.apiConfig.aggregator}/redirect_from`;
+    this.authParams = {
+      client_id: apiConfig.clientId,
+      client_secret: apiConfig.secret,
+    };
+  }
+
+  getOauthUrl(institutionId: string, state: string) {
+    return `${this.apiConfig.basePath}/auth?connector=${institutionId}&client_id=${this.apiConfig.clientId}&redirect_uri=${this.client_redirect_url}&state=${state}&response_type=code&scope=openid email profile offline_access`;
```

Comment:
Is `email` necessary in this scope? When I was testing locally, it didn't work until I removed `email`.